Sources/ComposableArchitecture/ViewStore.swift

```diff
@@ -83,9 +85,21 @@ public final class ViewStore<State, Action> {
     self.produced = Produced(by: producer)
     self.state = store.state
     self._send = store.send
-    producer.startWithValues { [weak self] in
-      self?.state = $0
-    }
+    let observer = Signal<State, Never>.Observer(
+      value: { [weak self] state in
+        self?.state = state
+      },
+      failed: .none,
+      completed: { [weak self] in
+        self?.viewDisposable?.dispose()
+        self?.viewDisposable = nil
+      },
+      interrupted: { [weak self] in
+        self?.viewDisposable?.dispose()
+        self?.viewDisposable = nil
+      }
+    )
+    viewDisposable = producer.start(observer)
```

Comment:
As above, why not just `viewDisposable = producer.starWithValues { ... }` ?

Reply:
I tried just using `startWithValues` and the new test you added passes, so I'm pretty sure it's equivalent, and makes the code much shorter and more readable. The only difference is that the disposable is not reset to `nil`, but that's not really an issue since calling `dispose()` on an already disposed `Disposable` does nothing.

Reply:
I removed actual tests for these leaks, because they can only be ran manually, but i suppose it should work without Observer instance, so feel free to remove it or I may do it tomorrow

Reply:
> so feel free to remove it

I'm not sure I can update your PR. I have it checked out locally in my own branch, but I can't push to the branch in your fork.

Reply:
Done ✅